packages/wallet/backend/src/gatehub/service.ts

```diff
@@ -69,9 +73,62 @@ export class GateHubService {
 
     await User.query().findById(user.id).patch(userDetails)
 
+    if (this.env.GATEHUB_ENV !== 'production') {
+      await this.setupNonProdUser(user.id, userState)
+    }
+
     return isUserApproved
   }
 
+  private async setupNonProdUser(
+    userId: string,
+    userState: IGetUserStateResponse
+  ) {
+    if (!this.accountService || !this.walletAddressService) {
+      throw new Error(
+        'AccountService and WalletAddressService must be provided in sandbox environment.'
+      )
+    }
+
+    const account = await this.accountService.createDefaultAccount(
+      userId,
+      'EUR Account'
+    )
+    if (!account) {
+      throw new Error('Failed to create account for managed user')
+    }
+
+    const walletAddressName: string = userState.meta['paymentPointer']
+    await this.walletAddressService.create({
+      userId,
+      accountId: account.userId,
+      walletAddressName,
+      publicName: walletAddressName
+    })
+
+    // Should only return one element in the array
+    const cardProducts = this.gateHubClient.fetchCardApplicationProducts()
+    if ((await cardProducts).length === 0) {
+      throw new Error('Failed to fetch product code')
+    }
+    const productCode = (await cardProducts).map((product) => product.code)
+
+    await this.gateHubClient.createCustomer({
+      walletAddress: account.gateHubWalletId,
+      account: {
+        productCode: productCode[0],
+        card: {
+          productCode: productCode[0],
+          nameOnCard: `INTERLEDGER $ILP.DEV/${walletAddressName}`
```

Comment:
Should we leave this hardcoded for now as we now the domain will be ilp.dev for sandbox?

Reply:
I would put in in an environment variable as well. This way you can reuse it in 
- https://github.com/interledger/testnet/pull/1675

since you are still creating the container there as wel..